compile_commands.json

```diff
@@ -1,45 +1,4 @@
 [
```

Comment:
might want to add this to the gitignore?

Reply:
nah it enables IntelliSense for anyone who clones the repo

Reply:
I think we don't want that so that we don't all have to individually create this file.